server/src/graql/internal/reasoner/atom/predicate/IdPredicate.java

```diff
@@ -18,65 +18,104 @@
 
 package grakn.core.graql.internal.reasoner.atom.predicate;
 
-import com.google.auto.value.AutoValue;
-import grakn.core.graql.internal.reasoner.atom.Atomic;
-import grakn.core.graql.internal.reasoner.query.ReasonerQuery;
 import grakn.core.graql.concept.Concept;
 import grakn.core.graql.concept.ConceptId;
 import grakn.core.graql.concept.Label;
 import grakn.core.graql.concept.SchemaConcept;
 import grakn.core.graql.exception.GraqlQueryException;
+import grakn.core.graql.internal.reasoner.atom.Atomic;
+import grakn.core.graql.internal.reasoner.query.ReasonerQuery;
+import grakn.core.graql.query.Query;
+import grakn.core.graql.query.pattern.property.IdProperty;
+import grakn.core.graql.query.pattern.property.ValueProperty;
 import grakn.core.graql.query.pattern.statement.Statement;
 import grakn.core.graql.query.pattern.statement.Variable;
-import grakn.core.graql.query.pattern.property.IdProperty;
-import grakn.core.graql.query.predicate.Predicates;
 import grakn.core.server.Transaction;
 
+import javax.annotation.CheckReturnValue;
+
 /**
- *
- * <p>
  * Predicate implementation specialising it to be an id predicate. Corresponds to {@link IdProperty}.
- * </p>
- *
- *
  */
-@AutoValue
-public abstract class IdPredicate extends Predicate<ConceptId>{
+public class IdPredicate extends Predicate<ConceptId> {
```

Comment:
Still think it's a perfect autovalue use case here. Instead we are removing one line and adding forty for no gain.